### PR TITLE
ROX-13593: Add telemetry options to central Helm chart

### PR DESCRIPTION
**Pull request by stehessel**
## Description

The current telemetry implementation has been deprecated, but not removed from the code base as a whole. It will be replaced by a new telemetry implementation (see #3819). We are planning to use an installation toggle to give opt-in/opt-out of the new telemetry collection. If enabled, the installation sets up a telemetry storage key, which enables the collection within Central.

The current `disableTelemetry` flag is removed. The current telemetry data store and api are deprecated and will be removed later. The installation toggle will be the only way to enable/disable telemetry - even if that then requires rolling the central pod.

This PR sets up the following configuration for the new telemetry stack:
- A user facing boolean setting to enable/disable telemetry (`telemetry.enabled`).
- Two options to configure the telemetry storage backend (`telemetry.storage`). Initially, at least the storage key has to be provided manually to enable telemetry. Eventually, the storage will be hardcoded with a default value in the chart.

An addition to the central CRD for the operator installation method will follow.

Note that we are not planning to roll out telemetry to on-premise users at this time. Telemetry will initially be a feature only enabled for the ACS cloud service. However, we want to lay the config ground work for future a on-premise rollout.

We opted to not document the flags publicly to avoid confusion of on-prem users, for whom the telemetry options are currently ineffective. The current `disableTelemetry` flag is still part of upstream documentation (https://docs.openshift.com/acs/3.72/installing/installing_helm/install-helm-customization.html#central-services-public-configuration-file-central_acs-install-helm-customization) - this entry will be removed from the documentation.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- [x] Evaluated and added CHANGELOG entry if required
- [ ] ~Determined and documented upgrade steps~
- [x] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### CODE REVIEW

**Review by dhaus67 — APPROVED** [review]
Sorry for taking so long to get to this PR, but LGTM! 🚢

**Comment by stehessel** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/4030?utm_source=stack-comment).


Job Reference: GCkBY1hkNIOUYR8puiWL

**Pull request merged**
